TemplePlus/ui/ui.cpp

```diff
@@ -790,6 +790,9 @@ bool UiManager::TranslateMouseMessage(const TigMouseMsg& mouseMsg)
 			bool enqueueExited = false;
 			auto globalWid = GetWidget(globalWidId);
 			// if window
+
+			if (globalWid == nullptr) return false;  //Happens sometimes on shutdown
```

Comment:
This probably also hides other bugs :-D

Reply:
This only happened on shutdown when TOEE is in windowed mode.  At this point it is probably OK that the window is NULL (it needs to get destroyed).  The mouse handler just needs to return.  This change prevents the crash and TOEE seems to shutdown successfully.

Reply:
If it's destroyed normally then it's better to directly check that the windows have been destroyed.

Reply:
Sounds good.  How do you check for a destroyed window (other than check for null pointer)?

Reply:
Just looked at the code.  It is checked for -1 before it gets into that part of the code.  It looks like GetWidget() just didn't find in in its widget list.

Reply:
If this is a spurious mouse move event triggering this after the UI has already been destroyed, the actual bug might be not unregistering the mouse move handler on the mainwindow.
I specifically mean MainWindow::SetMouseMoveHandler and the other event handling functions. You're right that these callbacks may trigger after the UI has already been destroyed, so they likely need to be unregistered.

Reply:
I think you are right and the cause probably is a spurious mouse move event.  However the handler gets created in the game view constructor and destroyed in the destructor so that looks ok.  Maybe game view is destroyed too late (it is destroyed at the end of GameLoop::Run()).

Reply:
Here is what the stack looks like.  It seems that the game loop has actually already been destroyed and the handler has been set to null.  Maybe The UiPcCreationSys destructor should check for a null game loop and not call mRace->Reset(uiPcCreation.GetCharEditorSelPacket()); if that is the case?

![image](https://user-images.githubusercontent.com/31427575/79059821-5b56df00-7c4c-11ea-80fd-4519ad765815.png)

Reply:
My suggestion above seems like it would indeed fix the problem but I am still not 100% sure it is the most correct solution.

Reply:
What we're actually missing here is something else, heh:
When the widget that `mWidgetMouseHandlerWidgetId` has the id of is removed, it should clear the field.
I.e.:

In `void UiManager::RemoveWidget(LgcyWidgetId id)`, the code should do something along those lines (at the function's end):
```
UnsetMouseCaptureWidgetId(mWidgetMouseHandlerWidgetId);
if (mMouseButtonId == id) {
mMouseButtonId = 0;
}
if (mWidgetMouseHandlerWidgetId == id) {
mWidgetMouseHandlerWidgetId = 0;
}
```

Reply:
This change may well be for the better but it doesn't fix the crash I am seeing.  When make your change and remove my fix, TOEE can still easily be crashed by pressing the "x" at the start menu.

Reply:
I'll look into it if it that's easy to repro :-D

Reply:
If you find a better way to prevent the crash on shutdown I think we can add it in a new pull request.  I don't think there is much more to say on this pull request so I am going to go ahead and merge it in.